language: python
python:
  - 2.7

dist:
 - xenial

sudo: false

addons:
  apt:
    sources:
      - sourceline: ppa:xapian-backports/ppa

    packages:
      # Required to build/install the xapian-binding
      - libxapian-dev
      # Required to install pyme
      - libgpgme11-dev
      - swig

before_install:
  - pip install sphinx==1.8.5
  - XAPIAN_VER=$(dpkg -l libxapian-dev | tail -n 1 | awk '{print $3}' | cut -d '-' -f 1)
  - cd /tmp
  - curl -s -O https://oligarchy.co.uk/xapian/$XAPIAN_VER/xapian-bindings-$XAPIAN_VER.tar.xz
  - tar -Jxvf xapian-bindings-$XAPIAN_VER.tar.xz
  - cd xapian-bindings-$XAPIAN_VER/
  - ./configure --prefix=$VIRTUAL_ENV --with-python && make && make install

  # change back to the checked out repository directory
  - cd $TRAVIS_BUILD_DIR

install:
  - pip install mysqlclient==1.3.13
  - pip install MySQL-python psycopg2 pytz whoosh
  # errors during build with:
  # make: *** No rule to make target '/gpgme.h', needed by 'gpgme.h'.  Stop.
  # ... gcc: error: gpgme_wrap.c: No such file or directory
  # so comment out for now and test will be skipped
  #- pip install git+https://bitbucket.org/malb/pyme.git
  - pip install pytest-cov codecov

before_script:
  # set up mysql database
  - ls -l /var/run/mysqld/mysqld.sock
  - sudo service mysql restart
  - sleep 60
  - mysql -u root -e 'GRANT ALL ON rounduptest.* TO rounduptest@localhost IDENTIFIED BY "rounduptest";'

  # set up postgresql database
  - psql -c "CREATE ROLE rounduptest WITH CREATEDB LOGIN PASSWORD 'rounduptest';" -U postgres

  # HACK: workaround mysql bug: http://bugs.mysql.com/bug.php?id=74901
  #   needed for test_mysql.mysqlDBTest.testFilteringSpecialChars
  - sed -i 's/CREATE DATABASE \%s/CREATE DATABASE \%s COLLATE utf8_general_ci/' roundup/backends/back_mysql.py

script:
  - py.test -v test/ --cov=roundup

after_success:
  - codecov
